gbpservice/nfp/configurator/agents/agent_base.py

```diff
@@ -0,0 +1,181 @@
+            return
+
+        # Multiple request data blobs needs batch processing. Send batch
+        # processing event or do direct processing of single request data blob
+        if (len(sa_info_list) > 1):
+            args_dict = {
+                'sa_info_list': sa_info_list,
+                'notification_data': notification_data
+            }
+            ev = self._sc.new_event(id='PROCESS_BATCH', data=args_dict)
+            self._sc.post_event(ev)
+        else:
+            sa_info_list[0]['context'].update(
+                {'notification_data': notification_data})
+            sa_info_list[0]['context'].update(
+                {'resource': sa_info_list[0]['resource']})
+            getattr(self, sa_info_list[0]['method'])(
+                sa_info_list[0]['context'],
+                **sa_info_list[0]['kwargs'])
+
+
+class AgentBaseEventHandler(object):
+
+    def __init__(self, sc, drivers, rpcmgr, nqueue):
+        self._sc = sc
+        self.drivers = drivers
+        self._rpcmgr = rpcmgr
+        self.nqueue = nqueue
+
+    def _notification(self, data):
+        event = self._sc.new_event(
+            id='NOTIFICATION_EVENT', key='NOTIFICATION_EVENT', data=data)
+        self._sc.poll_event(event)
+
+    def process_batch(self, ev):
+        """Processes a request with multiple data blobs.
+
+        Configurator processes the request with multiple data blobs and sends
+        a list of service information to be processed. This function goes
+        through the list of service information and invokes specific service
+        driver methods. After processing each request data blob, notification
+        data blob is prepared.
+
+        :param ev: Event instance that contains information of event type and
+        corresponding event data to be processed.
+
+        """
+
+        try:
+            # Get service agent information list and notification data list
+            # from the event data
+            sa_info_list = ev.data.get('sa_info_list')
+            notification_data = ev.data.get('notification_data')
+
+            # Process the first data blob from the service information list.
+            # Get necessary parameters needed for driver method invocation.
+            method = sa_info_list[0]['method']
+            resource = sa_info_list[0]['resource']
+            kwargs = sa_info_list[0]['kwargs']
+            request_info = kwargs['kwargs']['request_info']
+            del kwargs['kwargs']['request_info']
+            context = sa_info_list[0]['context']
+            service_type = kwargs.get('kwargs').get('service_type')
+
+            # Get the service driver and invoke its method
+            driver = self._get_driver(service_type)
+
+            # Service driver should return "success" on successful API
+            # processing. All other return values and exceptions are treated
+            # as failures.
+            result = getattr(driver, method)(context, **kwargs)
```

Comment:
Where is the processing for sa_info_list[1] and following. Why is this not a loop to process all the reqs?